mne/minimum_norm/resolution_matrix.py

```diff
@@ -0,0 +1,179 @@
+    info = deepcopy(inverse_operator['info'])
+    info['sfreq'] = 1000.  # necessary
+    info['projs'] = inverse_operator['projs']
+    return info
+
+
+def _get_matrix_from_inverse_operator(inverse_operator, forward, method='dSPM',
+                                      lambda2=1. / 9.):
+    """Get inverse matrix from an inverse operator.
+
+    Currently works only for fixed/loose orientation constraints
+    For loose orientation constraint, the CTFs are computed for the normal
+    component (pick_ori='normal').
+
+    Parameters
+    ----------
+    inverse_operator : instance of InverseOperator
+        The inverse operator.
+    forward : dict
+        The forward operator.
+    method : 'MNE' | 'dSPM' | 'sLORETA'
+        Inverse methods (for apply_inverse).
+    lambda2 : float
+        The regularization parameter (for apply_inverse).
+
+    Returns
+    -------
+    invmat : array, shape (n_dipoles, n_channels)
+        Inverse matrix associated with inverse operator and specified
+        parameters.
+    """
+    # make sure forward and inverse operators match
+    _convert_forward_match_inv(forward, inverse_operator)
+
+    info_inv = _prepare_info(inverse_operator)
+
+    # only use channels that are good for inverse operator and forward sol
+    ch_names_inv = info_inv['ch_names']
+    n_chs_inv = len(ch_names_inv)
+    bads_inv = inverse_operator['info']['bads']
+
+    # indices of bad channels
+    ch_idx_bads = [ch_names_inv.index(ch) for ch in bads_inv]
+
+    # create identity matrix as input for inverse operator
+    # set elements to zero for non-selected channels
+    id_mat = np.eye(n_chs_inv)
+
+    # convert identity matrix to evoked data type (pretending it's an epoch)
+    ev_id = EvokedArray(id_mat, info=info_inv, tmin=0.)
+
+    # apply inverse operator to identity matrix in order to get inverse matrix
+    # free orientation constraint not possible because apply_inverse would
+    # combine components
+
+    # check if inverse operator uses fixed source orientations
+    is_fixed_inv = inverse_operator['eigen_leads']['data'].shape[0] == \
+        inverse_operator['nsource']
+
+    # choose pick_ori according to inverse operator
+    if is_fixed_inv:
+        pick_ori = None
+    else:
+        pick_ori = 'vector'
+
+    # columns for bad channels will be zero
+    invmat_op = apply_inverse(ev_id, inverse_operator, lambda2=lambda2,
+                              method=method, pick_ori=pick_ori)
+
+    # turn source estimate into numpty array
+    invmat = invmat_op.data
+
+    dims = invmat.shape
+
+    # remove columns for bad channels
+    # take into account it may be 3D array
+    invmat = np.delete(invmat, ch_idx_bads, axis=len(dims) - 1)
+
+    # if 3D array, i.e. multiple values per location (fixed and loose),
+    # reshape into 2D array
+    if len(dims) == 3:
+        v0o1 = invmat[0, 1].copy()
+        v3o2 = invmat[3, 2].copy()
+        invmat = invmat.reshape(dims[0] * dims[1], dims[2])
```

Comment:
```suggestion
        shape = invmat.shape
        invmat = invmat.reshape(shape[0] * shape[1], shape[2])
```

then remove dims. I don't like the word `dims` as shape is the common name.